app/lib/search/updater.dart

```diff
@@ -4,92 +4,65 @@
-      return;
-    }
-    try {
-      _updating = true;
-      await _doUpdate(limit);
-    } catch (e, st) {
-      _logger.severe('Error while updating search index.', e, st);
-    } finally {
-      _updating = false;
+  @override
+  Stream<Task> pollTasks() {
+    final DateTime updatedAfter =
+        _isFirst ? null : new DateTime.now().subtract(new Duration(days: 1));
+    _isFirst = false;
+    Stream<Task> stream = searchBackend
+        .listPackages(updatedAfter: updatedAfter)
+        .map((String package) => new Task(package, null));
+    if (isLocalDevMode) {
+      // limit memory use and faster warm-up time
+      stream = stream.take(300);
     }
+    return stream;
   }
+}
 
-  Future _doUpdate(int limit) async {
-    final List<Key> versionKeys = [];
-    final Map<String, String> devVersions = {};
-
-    _logger.info('Polling packages for changes.');
-    int count = 0;
-    await for (Package p in dbService.query(Package).run()) {
-      final bool inIndex = await index.contains(
-          _toUrl(p.name), p.latestVersion, p.latestDevVersion);
-      if (inIndex) continue;
-      versionKeys.add(p.latestVersionKey);
-      devVersions[p.name] = p.latestDevVersion;
-      count++;
-      if (limit != null && limit <= count) break;
-    }
-    _logger.info('Found ${versionKeys.length} packages to update.');
-    if (versionKeys.isEmpty) return;
-
-    for (List<Key> slices in sliceList(versionKeys, 20)) {
-      final Key firstKey = slices.first;
-      final String firstPkg = firstKey.parent.id;
-      final String firstVersion = firstKey.id;
-      _logger.info('Updating packages staring with $firstPkg $firstVersion');
-      final List<PackageVersion> versions = await dbService.lookup(slices);
-      final List<PackageDocument> documents = versions
-          .map((pv) => new PackageDocument(
-                url: _toUrl(pv.package),
-                package: pv.package,
-                version: pv.version,
-                devVersion: devVersions[pv.package],
-                detectedTypes: pv.detectedTypes,
-                description: pv.pubspec.description,
-                lastUpdated: pv.shortCreated,
-                readme: pv.readmeContent,
-                popularity: mockScores[pv.package] ?? 0.0,
-              ))
-          .toList();
-      await index.addAll(documents);
-      _logger.info('Updated ${slices.length} packages.');
-    }
+// TODO: refactor task processing to support batches
+List<Task> _batch = [];
 
-    _logger.info('Calling index.merge() ...');
-    await index.merge();
-    _logger.info('index.merge() completed.');
+Future updateIndex(Task task) async {
+  _batch.add(task);
+  if (_batch.length >= 20) {
+    await _updateBatch();
   }
+}
 
-  void startPolling() {
-    if (_timer == null) {
-      _timer = new Timer.periodic(new Duration(hours: 4), (_) {
-        update();
-      });
-      update();
-    }
+Future mergeIndex(int taskCount) async {
```

Comment:
Consider adding a comment here: What is being merged? What does `taskCount` mean? Why is it not a boolean?